specs/merge/beacon-chain.md

```diff
@@ -59,148 +57,138 @@ We define the following Python custom types for type hinting and readability:
 
-The execution payload header included in a `BeaconState`.
-
-*Note:* Holds execution payload data without transaction bodies.
-
 ```python
 class ExecutionPayloadHeader(Container):
-    block_hash: Hash32  # Hash of execution block
+    # Execution block header fields
     parent_hash: Hash32
     coinbase: Bytes20
     state_root: Bytes32
-    number: uint64
+    logs_bloom: ByteVector[BYTES_PER_LOGS_BLOOM]
+    receipt_root: Bytes32
+    block_number: uint64
     gas_limit: uint64
     gas_used: uint64
     timestamp: uint64
-    receipt_root: Bytes32
-    logs_bloom: ByteVector[BYTES_PER_LOGS_BLOOM]
-    transactions_root: Root
+    # Extra payload fields
+    block_hash: Hash32  # Hash of execution block
+    opaque_transactions_root: Root
 ```
 
-## Protocols
-
-### `ExecutionEngine`
-
-The `ExecutionEngine` protocol separates the consensus and execution sub-systems.
-The consensus implementation references an instance of this sub-system with `EXECUTION_ENGINE`. 
-
-The following methods are added to the `ExecutionEngine` protocol for use in the state transition:
-
-#### `new_block`
+## Helper functions
 
-Verifies the given `execution_payload` with respect to execution state transition, and persists changes if valid.
+### Predicates
 
-The body of this function is implementation dependent.
-The Consensus API may be used to implement this with an external execution engine.
+#### `is_merge_complete`
 
 ```python
-def new_block(self: ExecutionEngine, execution_payload: ExecutionPayload) -> bool:
-    """
-    Returns True if the ``execution_payload`` was verified and processed successfully, False otherwise. 
-    """
-    ...
+def is_merge_complete(state: BeaconState) -> bool:
+    return state.latest_execution_payload_header != ExecutionPayloadHeader()
 ```
 
-## Helper functions
+#### `is_merge_block`
 
-### Misc
+```python
+def is_merge_block(state: BeaconState, body: BeaconBlockBody) -> bool:
+    return not is_merge_complete(state) and body.execution_payload != ExecutionPayload()
+```
 
 #### `is_execution_enabled`
 
 ```python
-def is_execution_enabled(state: BeaconState, block: BeaconBlock) -> bool:
-    return is_transition_completed(state) or is_transition_block(state, block)
+def is_execution_enabled(state: BeaconState, body: BeaconBlockBody) -> bool:
+    return is_merge_block(state, body) or is_merge_complete(state)
 ```
 
-#### `is_transition_completed`
+### Misc
 
-```python
-def is_transition_completed(state: BeaconState) -> bool:
-    return state.latest_execution_payload_header != ExecutionPayloadHeader()
-```
+#### `compute_timestamp_at_slot`
 
-#### `is_transition_block`
+*Note*: This function is unsafe with respect to overflows and underflows.
 
 ```python
-def is_transition_block(state: BeaconState, block: BeaconBlock) -> bool:
-    return not is_transition_completed(state) and block.body.execution_payload != ExecutionPayload()
+def compute_timestamp_at_slot(state: BeaconState, slot: Slot) -> uint64:
```

Comment:
`compute_time_at_slot` is already defined in phase 0 validator doc. I suggest

1. to not redefine it
2. to not change the name

Reply:
This function needs to be updated in merge/validator.md